test/model/test_requantize_mlp.py

```diff
@@ -0,0 +1,88 @@
+import io
+from tempfile import NamedTemporaryFile
+
+import pytest
+import torch
+from helpers import random_qactivation, random_tensor
+
+from quanto import (
+    Calibration,
+    QLinear,
+    QTensor,
+    freeze,
+    qint8,
+    quantize,
+    safe_load,
+    safe_save,
+)
+from quanto.nn import QModuleMixin
+from quanto.quantize import requantize
+
+
+class MLP(torch.nn.Module):
+    def __init__(self, input_size, output_size, hidden_size):
+        super().__init__()
+        self.input_layer = torch.nn.Linear(input_size, hidden_size)
+        self.mid_layer = torch.nn.Linear(hidden_size, hidden_size)
+        self.output_layer = torch.nn.Linear(hidden_size, output_size)
+
+    def forward(self, inputs):
+        x = torch.nn.functional.relu(self.input_layer(inputs))
+        x = torch.nn.functional.relu(self.mid_layer(x))
+        return torch.nn.functional.softmax(self.output_layer(x), dim=-1)
+
+
+def check_mlp(model, frozen):
+    assert isinstance(model.input_layer, QLinear)
+    assert isinstance(model.mid_layer, QLinear)
+    assert isinstance(model.output_layer, QLinear)
+    if frozen:
+        assert isinstance(model.input_layer.weight, QTensor)
+        assert isinstance(model.mid_layer.weight, QTensor)
+        assert isinstance(model.output_layer.weight, QTensor)
+
+
+def get_outputs(model, batch_size, input_features, device):
+    qinputs = random_qactivation((batch_size, input_features), dtype=torch.float32).to(device)
+    return model(qinputs)
+
+
+def save_and_reload_state_dict(state_dict, serialization):
+    if serialization == "safetensors":
+        with NamedTemporaryFile() as tmp_file:
+            safe_save(state_dict, tmp_file.name)
+            return safe_load(tmp_file.name)
+    else:
+        b = io.BytesIO()
+        torch.save(state_dict, b)
+        b.seek(0)
+        weights_only = serialization == "weights_only"
+        return torch.load(b, weights_only=weights_only)
+
+
+@pytest.mark.parametrize("weights", [qint8], ids=["w-qint8"])
+@pytest.mark.parametrize("dtype", [torch.float16, torch.float32], ids=["fp16", "fp32"])
+@pytest.mark.parametrize("serialization", ["weights_only", "pickle", "safetensors"])
+@pytest.mark.skip_device("cuda")
```

Comment:
Please do not skip the test on CUDA

Reply:
> Please do not skip the test on CUDA
I don't have cuda, that's the reason I skipped it.. I can remove that line and make a PR, but it fails locally in my laptop

Reply:
Removed skip test line and made a PR again

Reply:
![image](https://github.com/huggingface/quanto/assets/88357044/5b60d0b9-a5b7-4001-bfe5-acce6c5a64d5)
@dacorvo  Thanks, Now I understand rebase feature. Kindly  review this PR.